### PR TITLE
Update all simple cases of switching to `IWorkingBeatmap`

### PR DESCRIPTION
**Pull request by peppy**
These are all cases where thought should not be required, in theory.

- [x] Depends on #15632.
- [x] Depends on #15631.
- [x] Depends on #15594.

### CODE REVIEW

File: osu.Game.Tests/Visual/Navigation/TestScenePresentBeatmap.cs
Comment (smoogipoo) [inline osu.Game.Tests/Visual/Navigation/TestScenePresentBeatmap.cs:143]: Technically a change in behaviour, but I think it's fine as long as tests pass.

File: osu.Game.Tests/Visual/SongSelect/TestScenePlaySongSelect.cs
Comment (smoogipoo) [inline osu.Game.Tests/Visual/SongSelect/TestScenePlaySongSelect.cs:742]: Seems like this should use OnlineID? Might be worth going over similar cases in this file again since I'm not sure why some cases are doing `ID == ID` comparisons (e.g. `TestHideSetSelectsCorrectBeatmap`) when this PR seems to have changed them to use `OnlineID`.

**Review by smoogipoo — CHANGES_REQUESTED** [review]

**Comment by bdach** [comment]
Aside from the above comments (which I agree with), something weird is going on with the history here (contains unrelated commits already in `master`, but seemingly only with the committer changed).

File: osu.Game.Tests/Visual/Navigation/TestScenePresentBeatmap.cs
Comment (bdach) [reply osu.Game.Tests/Visual/Navigation/TestScenePresentBeatmap.cs:143]: I think the newly-returned `.Equals()` implementation could be applied to these and work correctly. But I'm not fussed either way.

Comment (peppy) [reply osu.Game.Tests/Visual/Navigation/TestScenePresentBeatmap.cs:143]: I intentionally changed this one. Felt safer.

**Comment by peppy** [comment]
> Aside from the above comments (which I agree with), something weird is going on with the history here (contains unrelated commits already in `master`, but seemingly only with the committer changed).

I'm highly unsure how that happened, I think I went through one too many rebases. Hopefully shouldn't look too bad when merged, and a non-issue? Can try force pushing again if it's preferred.

File: osu.Game.Tests/Visual/SongSelect/TestScenePlaySongSelect.cs
Comment (peppy) [reply osu.Game.Tests/Visual/SongSelect/TestScenePlaySongSelect.cs]: Have done  a pass of this file to only use `OnlineID` now.

**Comment by bdach** [comment]
> Hopefully shouldn't look too bad when merged, and a non-issue?

The committer field is included in the set of fields used to calculate the commit hashes, and because of that commits duplicated here have a different hash from the originals and will therefore appear twice in the full `git log` if merged as-is. However, *somehow* by some magic git doesn't seem to duplicate the commit entries when viewing the log or blame of a particular file. So I *think* any confusion should be minimal?

If you think it's best to fix it, though, rebasing this branch onto `master` and manually dropping all duplicated commits (so all except two, if I'm not mistaken) seems to do the trick.

**Comment by peppy** [comment]
Up to @smoogipoo , fixing is easy and I can do so as long as it isn't going to interrupt the review process.

**Comment by smoogipoo** [comment]
Well I'm ready to merge this PR otherwise, could you rebase to fix the history?

**Comment by peppy** [comment]
Has been done.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**
